t/Util.pm

```diff
@@ -587,14 +625,22 @@ sub spawn_forked {
     if ($pid) {
         close $cout;
         close $cerr;
-        my $guard = make_guard(sub {
+        my $kill = sub {
             return unless defined $pid;
+            if ($opts->{on_exit}) {
+                my ($out, $err);
+                $out = read_with_timeout($pin, 0.1);
+                $err = read_with_timeout($pin2, 0.1);
+                $opts->{on_exit}->($out, $err);
+            }
             kill 'TERM', $pid;
             while (waitpid($pid, 0) != $pid) {}
-        });
+            undef $pid;
+        };
         return +{
             pid => $pid,
-            kill => sub { undef $guard; },
+            kill => $kill,
+            guard => make_guard(sub { $kill->() }),
```

Comment:
Could you please elaborate why we need to make these changes?

IIUC, it is already possible with `spawn_forked` to wait for the child process to die then read stdout, stderr. You can do that by first calling `$child->{kill}->()` (which sends SIGTERM then waits for the process to exit), then reading from `$child->{stdout}` or `$child->{stderr}`.

To me it seems that `on_exit` callback is used for testing output after the child process exits. If that is the case, I think the existing mechanism should be sufficient.

Reply:
Fixed in e4b279455cec176b4b1af16e7d4415b2036a3e4e, thanks for the advice